### PR TITLE
Phase2-hgx136 Update hgcal geometry correcting recently reported bugs

### PR DESCRIPTION
**Pull request by bsunanda**
Also possibility to extend the geometry with varying slope in the lower edge

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/HGCalCommonData/data/hgcal/v9/hgcal.xml [hunk 1]
```diff
@@ -42,32 +42,33 @@
   <Constant name="radMixL13"             value="902.0*mm"/>
   <Constant name="radMixL14"             value="902.0*mm"/>
   <Constant name="radMixL15"             value="902.0*mm"/>
+  <Constant name="slope1"                value="[etaMax:slope]"/>
```
Comment (ianna) [inline Geometry/HGCalCommonData/data/hgcal/v9/hgcal.xml]: @bsunanda - there isn't much point in introducing an intermediate constant.

**Comment by ianna** [comment]
+1

**Comment by bsunanda** [comment]
@arunhep @lpernie @ggovi Please sign this PR. HGCal geometry parameters are not transmitted to DB yet. We shall do that once the geometry code stabilizes.

File: Geometry/HGCalCommonData/data/hgcal/v9/hgcal.xml (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/data/hgcal/v9/hgcal.xml]: We are planning alternative version of geometry. These changes are made in view of that. Also in future we would deviate from etaMax:slope

**Comment by lpernie** [comment]
+1

File: CondFormats/GeometryObjects/interface/PHGCalParameters.h [hunk 2]
```diff
@@ -65,7 +65,7 @@ class PHGCalParameters {
   std::vector<int>         layerGroupO_;
   std::vector<uint32_t>    trformIndex_;
   double                   waferR_;
-  double                   slopeMin_;
+  std::vector<double>      slopeMin_;
```
Comment (fabiocos) [inline CondFormats/GeometryObjects/interface/PHGCalParameters.h]: @bsunanda @ianna this CondFormats object encoding the geometry has not been used for DB payloads so far, is this correct? So this change does not break any backward compatibility

**Comment by bsunanda** [comment]
@fabiocos Please merge this PR

**Comment by fabiocos** [comment]
@bsunanda could you please reply to my question before?

File: CondFormats/GeometryObjects/interface/PHGCalParameters.h (hunk 2 above)
Comment (bsunanda) [reply CondFormats/GeometryObjects/interface/PHGCalParameters.h]: Yes it is correct. We have not put the Phase2 geometries in DB yet.

**Comment by bsunanda** [comment]
@fabiocos Please see my comment. This PR will not break any backward compatibility

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

@ggovi please see the discussion above, and comment further in case

**Pull request merged**